app/assets/javascripts/controllers/catalog/catalog_item_form_controller.js

```diff
@@ -451,12 +462,16 @@ ManageIQ.angular.app.controller('catalogItemFormController', ['$scope', '$timeou
   };
 
   // watch for all the drop downs on screen
-  'catalog provisioning_playbook retirement_playbook provisioning_machine_credential retirement_machine_credential provisioning_vault_credential retirement_vault_credential provisioning_network_credential retirement_network_credential provisioning_cloud_credential retirement_cloud_credential provisioning_dialog zone'.split(' ').forEach(idWatch);
+  'catalog currency provisioning_playbook retirement_playbook provisioning_machine_credential retirement_machine_credential provisioning_vault_credential retirement_vault_credential provisioning_network_credential retirement_network_credential provisioning_cloud_credential retirement_cloud_credential provisioning_dialog zone'.split(' ').forEach(idWatch);
 
   function idWatch(name) {
     var fieldName = 'vm._' + name;
     $scope.$watch(fieldName, function(value) {
       vm.catalogItemModel[name + '_id'] = value ? value.id : '';
+      if (name ==  'currency' && typeof value !== 'undefined') {
```

Comment:
`if (name == 'currency' && ! value)` should fix the problem when unsetting currency. (It goes from undefined to a currency, but then it goes to null, not undefined, when unselecting.)